pandas/core/arrays/integer.py

```diff
@@ -550,10 +549,19 @@ def cmp_method(self, other):
 
     def sum(self, skipna=True, min_count=0, **kwargs):
         nv.validate_sum((), kwargs)
-        result = masked_reductions.sum(
-            values=self._data, mask=self._mask, skipna=skipna, min_count=min_count
-        )
-        return result
+        return super()._reduce("sum", skipna=skipna, min_count=min_count)
+
+    def prod(self, skipna=True, min_count=0, **kwargs):
+        nv.validate_prod((), kwargs)
```

Comment:
why can't we pass mask?

Reply:
I thought this was cleaner because it removed code duplication between here and BaseMaskedArray (which is already using masked_reductions)

Reply:
oh ic, ok then. yeah let's merge this and FloatingArray, then pls de-duplicate what  you can.